environment

```diff
@@ -30,6 +30,13 @@ _select () {
 # https://github.com/boto/boto3/issues/1179#issuecomment-569940530
 #
 _preauth () {
+    if ! gcloud auth application-default print-access-token >/dev/null 2>&1; then
+        gcloud auth application-default login --project="$GOOGLE_PROJECT"
+    else
+        gcloud auth application-default set-quota-project "$GOOGLE_PROJECT"
+    fi
+    echo "Google application default credentials are in effect."
+    echo "Use 'gcloud auth application-default revoke' to remove them."
```

Comment:
Add this to `_deauth` instead and make the `print` statement at the end of `_preauth` more generic. It probably needs to move from the embeded Python to the containing shell code.